rm/rm-server/src/test/java/functionaltests/nodesrecovery/RecoverSSHInfrastructureV2Test.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    //@Test ATDD:  needs to find a way to live the node process alive
+    public void testRecoverLocalInfrastructureWithAliveNodes() throws Exception {
+        // kill the RM only by sending a SIGKILL and leave node processes alive
+        NodesRecoveryProcessHelper.findRmPidAndSendSigKill("RMStarterForFunctionalTest");
+        // nodes should be re-taken into account by the restarted RM
+        restartRmAndCheckFinalState(false);
+    }
+
+    @Test
+    public void testRecoverLocalInfrastructureWithDownNodes() throws Exception {
+        // kill the RM and also kill all node processes
+        rmHelper.killRM();
+        // nodes should be re-deployed by the restarted RM
+        restartRmAndCheckFinalState(true);
+    }
+
+    private void startRmWithConfig(String configurationFilePath) throws Exception {
+        String rmconf = new File(RMTHelper.class.getResource(configurationFilePath).toURI()).getAbsolutePath();
+        rmHelper.startRM(rmconf);
+        resourceManager = rmHelper.getResourceManager();
+    }
+
+    private void startRmAndCheckInitialState() throws Exception {
+        // start RM
+        startRmWithConfig(START_CONFIG);
+        Assert.assertTrue("Nodes preservation should be activated on RM start for the test",
+                          PAResourceManagerProperties.RM_PRESERVE_NODES_ON_EXIT.getValueAsBoolean());
+        assertThat(rmHelper.isRMStarted()).isTrue();
+
+        // check the initial state of the RM
+        assertThat(resourceManager.getState().getAllNodes().size()).isEqualTo(0);
+
+        resourceManager.createNodeSource(NODE_SOURCE_NAME,
+                                         SSHInfrastructureV2.class.getName(),
+                                         TestSSHInfrastructureV2.infraParams,
+                                         StaticPolicy.class.getName(),
+                                         TestSSHInfrastructureV2.policyParameters);
+        RMTHelper.waitForNodeSourceCreation(NODE_SOURCE_NAME,
+                                            TestSSHInfrastructureV2.NB_NODES,
+                                            this.rmHelper.getMonitorsHandler());
+
+        RMMonitorEventReceiver resourceManagerMonitor = (RMMonitorEventReceiver) resourceManager;
+        ArrayList<RMNodeSourceEvent> nodeSourceEvent = resourceManagerMonitor.getInitialState().getNodeSource();
+        assertThat(nodeSourceEvent.size()).isEqualTo(1);
+        assertThat(nodeSourceEvent.get(0).getSourceName()).isEqualTo(NODE_SOURCE_NAME);
+        assertThat(resourceManagerMonitor.getState().getAllNodes().size()).isEqualTo(TestSSHInfrastructureV2.NB_NODES);
+    }
+
+    private void restartRmAndCheckFinalState(boolean nodesShouldBeRecreated) throws Exception {
+        // restart RM
+        rmHelper = new RMTHelper();
+        startRmWithConfig(RESTART_CONFIG);
+        Assert.assertFalse("Nodes preservation should not be activated on RM restart for the test",
```

Comment:
Replace Assert.assertFalse by assertThat so you'll keep using the same library in all the class